### PR TITLE
Init command to add entries related to Haul in .gitignore.

### PR DESCRIPTION
**Pull request by ferrannp**

### CODE REVIEW

File: src/commands/init.js [hunk 1]
```diff
@@ -96,6 +96,25 @@ async function init() {
     }
   }
 
+  progress = ora('Adding new entries to .gitignore').start();
+
+  const pathToGitIgnore = path.join(cwd, '.gitignore');
+  if (fs.existsSync(pathToGitIgnore)) {
+    const _gitignore = fs.readFileSync(pathToGitIgnore);
```
Comment (satya164) [inline src/commands/init.js]: nit: remove underscore

File: src/commands/init.js [hunk 2]
```diff
@@ -96,6 +96,25 @@ async function init() {
...
+      const buffer = new Buffer('# Haul\n#\n.happypack\nhaul-debug.log\n\n');
+      fs.writeSync(fd, buffer, 0, buffer.length, 0);
+      fs.writeSync(fd, _gitignore, 0, _gitignore.length, buffer.length);
+      fs.close(fd);
+      progress.succeed('Added new entries to .gitignore');
+    } else {
+      progress.info('.gitignore already contains # Haul');
+    }
+  } else {
+    progress.warn('No .gitignore found, skipping adding new entries');
```
Comment (satya164) [inline src/commands/init.js]: `No .gitignore found, skipping`

File: src/commands/init.js [hunk 3]
```diff
@@ -96,6 +96,25 @@ async function init() {
     }
   }
 
+  progress = ora('Adding new entries to .gitignore').start();
+
+  const pathToGitIgnore = path.join(cwd, '.gitignore');
+  if (fs.existsSync(pathToGitIgnore)) {
+    const _gitignore = fs.readFileSync(pathToGitIgnore);
+    if (!_gitignore.includes('# Haul')) {
```
Comment (satya164) [inline src/commands/init.js]: prolly should check for individual entries rather than the comment?

File: src/commands/init.js [hunk 4]
```diff
@@ -96,6 +96,25 @@ async function init() {
...
+      const fd = fs.openSync(pathToGitIgnore, 'w+');
+      const buffer = new Buffer('# Haul\n#\n.happypack\nhaul-debug.log\n\n');
+      fs.writeSync(fd, buffer, 0, buffer.length, 0);
```
Comment (satya164) [inline src/commands/init.js]: nit: maybe a new line here for better readability

File: src/commands/init.js (hunk 3 above)
Comment (satya164) [inline src/commands/init.js]: nit: new line before

File: src/commands/init.js (hunk 2 above)
Comment (grabbou) [inline src/commands/init.js]: Maybe we create it if it's missing?

File: src/commands/init.js [hunk 5]
```diff
@@ -96,6 +96,25 @@ async function init() {
...
+      fs.writeSync(fd, _gitignore, 0, _gitignore.length, buffer.length);
+      fs.close(fd);
+      progress.succeed('Added new entries to .gitignore');
+    } else {
+      progress.info('.gitignore already contains # Haul');
```
Comment (grabbou) [inline src/commands/init.js]: I'd rephrase that to something better, e.g. `Haul files are already ignored`

File: src/commands/init.js [hunk 6]
```diff
@@ -96,6 +96,25 @@ async function init() {
     }
   }
 
+  progress = ora('Adding new entries to .gitignore').start();
```
Comment (grabbou) [inline src/commands/init.js]: How about we show this only if: a) `gitignore` exists and it doesn't contain `haul` yet? Otherwise it feels like unnecessary output.

File: src/commands/init.js (hunk 2 above)
Comment (satya164) [inline src/commands/init.js]: In that case, we should detect if it's a git project first

File: src/commands/init.js (hunk 3 above)
Comment (grabbou) [inline src/commands/init.js]: +1

File: src/commands/init.js (hunk 2 above)
Comment (ferrannp) [inline src/commands/init.js]: In theory, it should be there if its a RN app. If it is not, that is probably dev decision so I would not create it or anything.

Comment (satya164) [inline src/commands/init.js]: Sounds fair

File: src/commands/init.js [hunk 7]
```diff
@@ -96,6 +96,25 @@ async function init() {
...
+
+  const pathToGitIgnore = path.join(cwd, '.gitignore');
+  if (fs.existsSync(pathToGitIgnore)) {
+    const _gitignore = fs.readFileSync(pathToGitIgnore);
+    if (!_gitignore.includes('# Haul')) {
+      const fd = fs.openSync(pathToGitIgnore, 'w+');
+      const buffer = new Buffer('# Haul\n#\n.happypack\nhaul-debug.log\n\n');
```
Comment (satya164) [inline src/commands/init.js]: Why buffer instead of just appending the text to what we read above?

Comment (ferrannp) [inline src/commands/init.js]: To add it at the beginning of the file, not at the end.

Comment (satya164) [inline src/commands/init.js]: Then you can just prepend, no? Though I think appending is better

```
fs.writeFileSync(dedent`${gitignore}

# Haul
haul-debug.log
.happypack
`, pathToFile);

File: src/commands/init.js [hunk 8]
```diff
@@ -96,23 +96,33 @@ async function init() {
...
-    if (!_gitignore.includes('# Haul')) {
-      const fd = fs.openSync(pathToGitIgnore, 'w+');
-      const buffer = new Buffer('# Haul\n#\n.happypack\nhaul-debug.log\n\n');
-      fs.writeSync(fd, buffer, 0, buffer.length, 0);
-      fs.writeSync(fd, _gitignore, 0, _gitignore.length, buffer.length);
-      fs.close(fd);
-      progress.succeed('Added new entries to .gitignore');
+    const gitignore = fs.readFileSync(pathToGitIgnore);
+
+    if (!gitignore.includes('# Haul')) {
```
Comment (satya164) [inline src/commands/init.js]: `gitignore.split('\n').includes`

**Comment by grabbou** [comment]
What's the status of this PR?

**Comment by ferrannp** [comment]
I need to finish with the split stuff, I'll let you know.
